R/caretList.R

```diff
@@ -322,3 +341,16 @@ predict.caretList <- function(object, newdata = NULL, ..., verbose = FALSE){
 
   return(preds)
 }
+
+#' @title Index a caretList
+#' @description Index a caret list to extract caret models into a new caretList object
+#"
+#' @param object an object of class caretList
+#' @export
+`[.caretList` <- function(object, index){
+  newObject <- object
+  class(newObject) <- "list"
+  newObject <- newObject[index]
+  class(newObject) <- "caretList"
+  return(newObject)
```

Comment:
I feel like there must be something we can do here to just use the `[` method from `list`.  Maybe `newObject <- [.listof`(object, index)`?